src/components/WalletDropdown/TransactionMenu.tsx

```diff
@@ -1,9 +1,159 @@
 import { Trans } from '@lingui/macro'
+import { useWeb3React } from '@web3-react/core'
+import { getYear, isSameDay, isSameWeek, isSameYear } from 'date-fns'
+import { useCallback, useMemo } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import styled from 'styled-components/macro'
 
+import { useAllTransactions } from '../../state/transactions/hooks'
+import { clearAllTransactions } from '../../state/transactions/reducer'
+import { TransactionDetails } from '../../state/transactions/types'
+import { TransactionSummary } from '../AccountDetailsV2'
 import { SlideOutMenu } from './SlideOutMenu'
 
-export const TransactionHistoryMenu = ({ onClose }: { onClose: () => void }) => (
-  <SlideOutMenu onClose={onClose} onClear={undefined} title={<Trans>Transactions</Trans>}>
-    <div />
-  </SlideOutMenu>
-)
+const Divider = styled.div`
+  margin-top: 16px;
+  border-bottom: ${({ theme }) => `1px solid ${theme.backgroundOutline}`};
+`
+
+const TransactionListWrapper = styled.div`
+  ${({ theme }) => theme.flexColumnNoWrap};
+`
+
+interface TransactionInformation {
+  title: string
+  transactions: TransactionDetails[]
+}
+
+const TransactionTitle = styled.span`
+  padding-bottom: 8px;
+  padding-top: 20px;
+  padding-left: 12px;
+  padding-right: 12px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.textTertiary};
+`
+
+const renderTransactions = (transactionInformation: TransactionInformation) => {
+  const { title, transactions } = transactionInformation
+
+  return (
+    <TransactionListWrapper key={title}>
+      <TransactionTitle>{title}</TransactionTitle>
+      {transactions.map((transactionDetails, i) => (
+        <TransactionSummary key={i} transactionDetails={transactionDetails} />
+      ))}
+    </TransactionListWrapper>
+  )
+}
+
+const getConfirmedTransactions = (confirmedTransactions: Array<TransactionDetails>) => {
+  const ONE_DAY = 1000 * 60 * 60 * 24
+  const THIRTY_DAYS = ONE_DAY * 30
+  const NOW = new Date().getTime()
+
+  const TODAY: Array<TransactionDetails> = []
```

Comment:
this should not be capitalized as they are not constants. I would also name it `today_transactions`